scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -973,16 +973,61 @@
                                        {:job-uuid->reserved-host (apply dissoc job-uuid->reserved-host matched-job-uuids)
                                         :launched-job-uuids (into matched-job-uuids launched-job-uuids)})))
 
+(defn job->preferred-compute-clusters
+  "Given a job and a collection of compute clusters, returns the
+  subset of compute clusters that the job would prefer to run
+  (and therefore, autoscale) on"
+  [{:keys [job/checkpoint job/instance]} compute-clusters]
+  (if (and checkpoint instance)
+    ; If checkpointing is enabled, we want to run the new instance in the
+    ; same location as the checkpointed instance to take advantage of data
+    ; locality for the checkpoint data
+    (let [{{:keys [compute-cluster/cluster-name]} :instance/compute-cluster}
+          (->> instance
+               (sort-by :instance/start-time)
+               last)
+          ; This is a list, but it should never
```

Comment:
maybe delete this "previous-compute-clusters" and replace with if-let

(let [{{:keys [compute-cluster/cluster-name]} :instance/compute-cluster}
      (->> instance
           (sort-by :instance/start-time)
           last)
      compute-cluster->location #(-> %
                                   :cluster-definition
                                   :config
                                   :location)]  
  (if-let [previous-location
           (->> compute-clusters
                (filter #(= (:name %) cluster-name))
                first
                compute-cluster->location)]
    (filter
      #(= (compute-cluster->location %) previous-location)
      compute-clusters)
    ; If the previous instance's compute cluster name is not
    ; present in the list of compute clusters passed, there's
    ; not much we can do
    compute-clusters))

Reply:
Good call, done